modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -619,12 +616,26 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
                     Assignment.Status.EXPERIMENT_NOT_STARTED);
         }
 
+        Assignment assignment = getAssignment(experimentID, userID, context, userAssignments, bucketList);
+
+        // Case: Experiment expires -> existing Assignments are returned and new Assignments
+        // are returned as null assignments
         if (currentTime > experiment.getEndTime().getTime()) {
-            return nullAssignment(userID, applicationName, experimentID,
-                    Assignment.Status.EXPERIMENT_EXPIRED);
+
+            // the experiment moves to the PAUSED state if it was RUNNING before
+            if (Experiment.State.RUNNING == experiment.getState()) {
```

Comment:
@AndreaSuckro In addition to the in-memory experiment state check (should be RUNNING in memory), you need to add MySQL experiment state check (should be RUNNING in MySQL) to avoid multiple emails in a thread race condition - similar to the isAssignmentValidForRapidExperiment() method...